### ISSUE TITLE
Encapsulate damaging player inside RdPlayer class

### ISSUE THREAD

**Issue by David-Estevez** [asrob-uc3m/robotDevastation#23]
Currently there is no explicit way of damaging a player. It involves retrieving the current level of heath and setting the new one. 

I think this should be encapsulated.

**Comment by David-Estevez** [comment]
There is also no way of restoring health of player through mentalMap. I had to do nasty things to solve this when respawning dead player:

```c++
        //-- Restore things (health, enable stuff)
        //-- Note: this is a workaround because one cannot modify player's health through
        //-- mentalMap directly (yet)
        std::vector<RdPlayer> players = mentalMap->getPlayers();
        int id = mentalMap->getMyself().getId();
        for (int i = 0; i < players.size(); i++)
            if (players[i].getId() == id)
            {
                players[i].setHealth(MAX_HEALTH);
                break;
            }
        mentalMap->updatePlayers(players);
```

**Comment by David-Estevez** [comment]
To solve this, I propose the following addition to the MentalMap API:
```cpp
class MentalMap
{
    //-- Restores the health of current player (and does more stuff if needed)
    bool respawn();
}
```

And the following addition to RdPlayer API:

```cpp
class RdPlayer
{
   //-- Reduces this player's health by an amount specified by the weapon used
   bool getDamageFromWeapon(RdWeapon& weapon);
}
```

**Comment by jgvictores** [comment]
Seems reasonable.

**Comment by David-Estevez** [comment]
Fixed in 469edeb
